ImagePicker/CarvedLabel.swift

```diff
@@ -63,51 +66,35 @@ final class CarvedLabel : UIView {
         isOpaque = false
     }
     
-    override var backgroundColor: UIColor? {
-        get { return UIColor.clear }
-        set { super.backgroundColor = UIColor.clear }
-    }
-    
-    fileprivate var textAttributes: TextAttributes {
-        let activeFont = font ?? UIFont.systemFont(ofSize: 12, weight: .regular)
-        return [
-            NSAttributedStringKey.font: activeFont
-        ]
-    }
-    
-    fileprivate var attributedString: NSAttributedString {
-        return NSAttributedString(string: text ?? "", attributes: textAttributes)
-    }
-    
     override func draw(_ rect: CGRect) {
         let color = tintColor!
         color.setFill()
         
         let path = UIBezierPath(roundedRect: rect, cornerRadius: cornerRadius)
         path.fill()
         
-        guard let context = UIGraphicsGetCurrentContext(), (text?.count ?? 0) > 0 else {
-            return
-        }
-        
-        let attributedString = self.attributedString
-        let stringSize = attributedString.size()
-        
-        let xOrigin: CGFloat = max(horizontalInset, (rect.width - stringSize.width)/2)
-        let yOrigin: CGFloat = max(verticalInset, (rect.height - stringSize.height)/2)
-        
-        context.saveGState()
-        context.setBlendMode(.destinationOut)
-        attributedString.draw(at: CGPoint(x: xOrigin, y: yOrigin))
-        context.restoreGState()
+        drawText(rect)
     }
     
     override func sizeThatFits(_ size: CGSize) -> CGSize {
         let stringSize = attributedString.size()
-        return CGSize(width: stringSize.width + horizontalInset*2, height: stringSize.height + verticalInset*2)
+        return CGSize(width: stringSize.width + horizontalInset * 2, height: stringSize.height + verticalInset * 2)
     }
 
     override var intrinsicContentSize: CGSize {
         return sizeThatFits(.zero)
     }
+    
+    private func drawText(_ rect: CGRect) {
+        guard let context = UIGraphicsGetCurrentContext(), text?.count ?? 0 > 0 else { return }
```

Comment:
add let textCount to guard to simplify the understanding of the condition. Make the comparison as the last one.

Reply:
Done